### PR TITLE
chore(core): type hinted various lib functions return types

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: docs/appendix/upgrade-notes/4.x-to-5.0.rst [hunk 1]
```diff
@@ -52,13 +52,17 @@ Lib functions return types
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 * ``elgg_add_admin_notice()`` now returns ``null`` on failure instead of a ``bool``
+* ``elgg_delete_metadata()`` no longer returns ``null``
 * ``elgg_get_annotation_from_id()`` now returns ``null`` if no annotation is found instead of a ``bool``
 * ``elgg_get_download_url()`` now returns ``null`` if there is an error
 * ``elgg_get_entity_as_row()`` now returns ``null`` if no entity is found instead of a ``bool``
 * ``elgg_get_entity_dates()`` no longer returns a ``bool``
 * ``elgg_get_inline_url()`` now returns ``null`` if there is an error
+* ``elgg_get_metadata_from_id()`` now returns ``null`` no metadata could be found
```
Comment (jdalsem) [inline docs/appendix/upgrade-notes/4.x-to-5.0.rst:61]: if not metadata

File: engine/classes/ElggEntity.php [hunk 2]
```diff
@@ -489,24 +489,23 @@ protected function setTempMetadata($name, $value, $multiple = false) {
 	 * @warning Calling this with no $name will clear all metadata on the entity.
 	 *
 	 * @param null|string $name The name of the metadata to remove.
-	 * @return bool|null
+	 *
+	 * @return bool
 	 * @since 1.8
 	 */
-	public function deleteMetadata($name = null) {
+	public function deleteMetadata($name = null): bool {
```
Comment (jdalsem) [inline engine/classes/ElggEntity.php:496]: should we make a deleteAllMetadata entity function to make deleting all metadata more explicit?

File: engine/lib/navigation.php [hunk 3]
```diff
@@ -123,8 +114,8 @@ function elgg_register_menu_item($menu_name, $menu_item) {
  * @return \ElggMenuItem|null
  * @since 1.8.0
  */
-function elgg_unregister_menu_item($menu_name, $item_name) {
-	$menus = _elgg_services()->config->menus;
+function elgg_unregister_menu_item(string $menu_name, string $item_name): ?\ElggMenuItem {
```
Comment (jdalsem) [inline engine/lib/navigation.php:117]: do we really want this to return a menu item?

**Review by jdalsem — APPROVED** [review]

File: engine/lib/navigation.php (hunk 3 above)
Comment (jeabakker) [reply engine/lib/navigation.php:117]: maybe you wanted to remove the menu item from menu 'a' and register it to menu 'b' this allows that

File: engine/classes/ElggEntity.php (hunk 2 above)
Comment (jeabakker) [reply engine/classes/ElggEntity.php:496]: maybe, but not in this PR

File: mod/groups/views/default/resources/groups/search.php [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 elgg_push_breadcrumb(elgg_echo('groups'), elgg_generate_url('collection:group:group:all'));
 
-$tag = get_input('tag');
+$tag = get_input('tag', '');
```
Comment (jdalsem) [inline mod/groups/views/default/resources/groups/search.php:5]: cast required

**Review by jdalsem — APPROVED** [review]

**Review by jdalsem — APPROVED** [review]

**Pull request merged**